packages/react/src/components/Button/Button.tsx

```diff
@@ -69,7 +74,8 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
           data-size={size}
           data-variant={variant}
           ref={ref}
-          type={type}
+          /* don't set type when we use `asChild` */
+          {...(asChild ? { asChild: true } : { type: 'button' })}
```

Comment:
Should we here have
```suggestion
          {...(asChild ? {asChild: true, type } : { type: 'button' })}
```
so that _when_ type is passed to `Button type="button" asChild` it is still passed down? ☺️

Reply:
oh doh, it should not set the default here :P Yes this works fine if we don't set the default when getting props